Automattic/wp-calypso
# Logging domainCart dependency error instead of throwing it on production

Related to 4150-gh-Automattic/dotcom-forge

## Proposed Changes

* Instead of showing the technical error to user, we're using a generic one:

![image](https://github.com/Automattic/wp-calypso/assets/1044309/9ef0a03f-c947-4386-83ad-a1624b370189)


## Testing Instructions



* Comment the lines 447 and 458 [of this file](https://github.com/Automattic/wp-calypso/blob/trunk/client/signup/steps/domains/index.jsx#L447)
* Simulate the error by following this video: p1696888640525179/1696866404.150999-slack-CKZHG0QCR
* Ensure you see a logstash payload with the error message:

![image](https://github.com/Automattic/wp-calypso/assets/1044309/3b2a286b-680d-431c-84f1-2ef7d45bee5a)

* Ensure you find the logs:

![image](https://github.com/Automattic/wp-calypso/assets/1044309/a387d4e2-b098-4e90-b171-6c898501e7f4)

* Ensure [the logic](https://github.com/Automattic/wp-calypso/blob/update/change-calypso-dependency-error/client/lib/signup/flow-controller.ts#L301-L303) make sense to only show the error on non production environment.

## Pre-merge Checklist
